File: MegaHerdt.Models/Models/Bill.cs
﻿
using System.ComponentModel.DataAnnotations;

namespace MegaHerdt.Models.Models
{
    public class Bill
    {
        [Key]
        public int Id { get; set; }
        public int? ReparationId { get; set; }
        public int? PurchaseId { get; set; }
        public string Type { get; set; }
        [StringLength(5)]
        public string SaleNumber { get; set; }
        [StringLength(8)]
        public string Number { get; set; }
        public Reparation? Reparation { get; set; }
        public Purchase? Purchase { get; set; }
        public List<Payment> Payments { get; set; }
    }
}

File: MegaHerdt.FrontEnd/MegaHerdt/src/app/models/Bill/Bill.ts
import { Payment } from "../Payment/Payments";

export class Bill{
    id: number;
    number: string;
    saleNumber: string;
    type: string;
    payments: Array<Payment>;
    constructor(id:number, saleNumber: string,number: string, type: string){
        this.id=id;
        this.saleNumber=saleNumber;
        this.number=number;
        this.type = type;
        this.payments = [];
    }
}